pkg/agent/plugin/workloadattestor/k8s/k8s_test.go

```diff
@@ -72,30 +85,102 @@ func TestK8s_AttestInitPidInPod(t *testing.T) {
-	mockHttpClient.EXPECT().Get("http://localhost:"+kubeletReadOnlyPort+"/pods").Return(
+	mockHttpClient.EXPECT().Get(podsURL).Return(
 		&http.Response{
-			Body: ioutil.NopCloser(bytes.NewReader(podList)),
+			StatusCode: http.StatusOK,
+			Body:       ioutil.NopCloser(bytes.NewReader(podList)),
 		}, nil)
 
 	mockFilesystem := filesystem_mock.NewMockfileSystem(mockCtrl)
-	mockFilesystem.EXPECT().Open(fmt.Sprintf("/proc/%v/cgroup", pid)).Return(os.Open(cgInitPidInPodFilePath))
+	mockFilesystem.EXPECT().Open(pidCgroupPath).Return(os.Open(cgInitPidInPodFilePath))
 
-	plugin, _, err := PluginGenerator(validConfig, mockHttpClient, mockFilesystem)
+	plugin := InitPlugin(t, mockHttpClient, mockFilesystem)
 	req := workloadattestor.AttestRequest{Pid: int32(pid)}
 	resp, err := plugin.Attest(&req)
 	require.NoError(t, err)
 	require.NotEmpty(t, resp.Selectors)
 }
 
+func TestK8s_AttestPidInPodAfterRetry(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	podList, err := ioutil.ReadFile(podListFilePath)
+	require.NoError(t, err)
+
+	podListNotRunning, err := ioutil.ReadFile(podListNotRunningFilePath)
+	require.NoError(t, err)
+
+	mockHttpClient := http_client_mock.NewMockhttpClient(mockCtrl)
+	mockHttpClient.EXPECT().Get(podsURL).Return(
+		&http.Response{
+			StatusCode: http.StatusOK,
+			Body:       ioutil.NopCloser(bytes.NewReader(podListNotRunning)),
+		}, nil)
+	mockHttpClient.EXPECT().Get(podsURL).Return(
+		&http.Response{
+			StatusCode: http.StatusOK,
+			Body:       ioutil.NopCloser(bytes.NewReader(podListNotRunning)),
+		}, nil)
+	mockHttpClient.EXPECT().Get(podsURL).Return(
+		&http.Response{
+			StatusCode: http.StatusOK,
+			Body:       ioutil.NopCloser(bytes.NewReader(podList)),
+		}, nil)
+
+	mockFilesystem := filesystem_mock.NewMockfileSystem(mockCtrl)
+	mockFilesystem.EXPECT().Open(pidCgroupPath).Return(os.Open(cgPidInPodFilePath))
+
+	plugin := InitPlugin(t, mockHttpClient, mockFilesystem)
+	req := workloadattestor.AttestRequest{Pid: int32(pid)}
+	resp, err := plugin.Attest(&req)
+	require.NoError(t, err)
+	require.NotEmpty(t, resp.Selectors)
+}
+
+func TestK8s_AttestPidNotInPodAfterRetry(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	podListNotRunning, err := ioutil.ReadFile(podListNotRunningFilePath)
+	require.NoError(t, err)
+
+	mockHttpClient := http_client_mock.NewMockhttpClient(mockCtrl)
+	mockHttpClient.EXPECT().Get(podsURL).Return(
+		&http.Response{
+			StatusCode: http.StatusOK,
+			Body:       ioutil.NopCloser(bytes.NewReader(podListNotRunning)),
+		}, nil)
+	mockHttpClient.EXPECT().Get(podsURL).Return(
+		&http.Response{
+			StatusCode: http.StatusOK,
+			Body:       ioutil.NopCloser(bytes.NewReader(podListNotRunning)),
+		}, nil)
+	mockHttpClient.EXPECT().Get(podsURL).Return(
+		&http.Response{
+			StatusCode: http.StatusOK,
+			Body:       ioutil.NopCloser(bytes.NewReader(podListNotRunning)),
+		}, nil)
```

Comment:
You can tack on `.Times(3)` here to avoid the repeated EXPECT() declarations

Reply:
Neat!

Reply:
Turns out `.Times()` doesn't work out too well without other changes. The same response object is returned (and the body has already been consumed after the first call).